mautic/mautic
# Add delivered count to email stats

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds delivered counter to email stats. 

Delivered = Sent - Bounced - Failed

![image](https://user-images.githubusercontent.com/8580942/218481679-0dd57986-8202-421f-988d-881afd5395d4.png)




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create an email 
3. Send to contacts and generate some hard bounces* or failed emails
4. View the email page and check the delivered emails counter
5. Create an email with translations and repeat steps 3 and 4
6. Create an email with AB tests and repeat steps 3 and 4

\* I've used Mailhog to send emails and then switched to Amazon API (credentials are not required) and used a webhook to record some bouces. You have to replace email address in this payload and X-EMAIL-ID header value.
```
POST /mailer/amazon_api/callback
{
  "eventType": "Bounce",
  "bounce": {
    "feedbackId": "010207655655bd3d-31765456-e7c7-4442-65454bd75c6545647-000000",
    "bounceType": "Permanent",
    "bounceSubType": "General",
    "bouncedRecipients": [
      {
        "emailAddress": "michael.jones.280@example.com",
        "action": "failed",
        "status": "5.0.0",
        "diagnosticCode": "smtp; 550 Message was not accepted -- invalid mailbox.  Local mailbox is unavailable: user not found"
      }
    ],
    "timestamp": "2023-02-02T21:37:43.000Z",
    "reportingMTA": "dns; example.i"
  },
  "mail": {
    "timestamp": "2023-02-02T09:40:21.755Z",
    "source": "\"Example\" <sender@example.com>",
    "sourceArn": "arn:aws:ses:eu-west-1:6666666666666:identity/sender@example.com",
    "sourceIp": "127.0.0.1",
    "callerIdentity": "ses-api-mautic-prod",
    "sendingAccountId": "756434343333",
    "messageId": "0102018643543e7b-1c367b9f-15c9-45f1-b8a0-8543543301e1-000000",
    "destination": [
      "michael.jones.280@example.com"
    ],
    "headersTruncated": false,
    "headers": [
      {
        "name": "Return-Path",
        "value": "<no-reply@example.com>"
      },
      {
        "name": "Message-ID",
        "value": "<bcf9238755435435435435435410d0c3@swift.generated>"
      },
      {
        "name": "Date",
        "value": "Thu, 02 Feb 2023 10:40:21 +0100"
      },
      {
        "name": "Subject",
        "value": "Be Part of the Business 4.0 Revolution"
      },
      {
        "name": "From",
        "value": "Example <sender@example.com>"
      },
      {
        "name": "Reply-To",
        "value": "sender@example.com"
      },
      {
        "name": "To",
        "value": "michael.jones.280@example.com"
      },
      {
        "name": "X-EMAIL-ID",
        "value": "18"
      }
    ],
    "commonHeaders": {
      "returnPath": "no-reply@example.com",
      "from": [
        "Example <sender@example.com>"
      ],
      "replyTo": [
        "sender@example.com"
      ],
      "date": "Thu, 02 Feb 2023 10:40:21 +0100",
      "to": [
        "michael.jones.280@example.com"
      ],
      "messageId": "<bcf92370afb2048f6a5ea3311710d0c3@swift.generated>",
      "subject": "Be Part of the Business 4.0 Revolution"
    }
  }
}
```
